engine/src/main/scala/cromwell/webservice/WorkflowSubmitBatchActor.scala

```diff
@@ -0,0 +1,47 @@
+package cromwell.webservice
+
+import akka.actor.{Actor, ActorRef, Props}
+import cromwell.engine.WorkflowSourceFiles
+import cromwell.engine.workflow.WorkflowManagerActor
+import cromwell.webservice.CromwellApiHandler.{WorkflowManagerBatchSubmitResponse, WorkflowManagerResponse}
+
+/** Submits a sequence of sources, then messages back the respective sequence of responses. */
+class WorkflowSubmitBatchActor(responseActor: ActorRef, requestHandlerActor: ActorRef,
+                               sources: Seq[WorkflowSourceFiles]) extends Actor {
+
+  private val responses = Array.fill[Option[WorkflowManagerResponse]](sources.size)(None)
+
+  override def preStart() = {
+    sources.zipWithIndex foreach {
+      case (source, index) =>
+        context.actorOf(
+          Props(new WorkflowSubmitIndexedActor(self, requestHandlerActor, source, index)),
+          "WorkflowSubmitIndexedActor_" + index)
+    }
```

Comment:
Oh ok I understand, I thought the main reason for the `WorkflowSubmitIndexedActor` was that the `WorkflowSubmitBatchActor` could start filling its array as soon as the first workflow is sent, but it's actually for the indexing. 👍